### PR TITLE
3DS: 2 bugfixes, 1 feature, a bit of cleanup

### PR DESCRIPTION
**Pull request by carstene1ns**
ttt

### CODE REVIEW

File: src/platform/3ds/audio.cpp [hunk 1]
```diff
@@ -208,164 +238,158 @@ void CtrAudio::SE_Play(std::unique_ptr<AudioSeCache> se, int volume, int pitch)
...
+	DSP_FlushDataCache(se.buf[chan].data_pcm16, aligned_bsize);
 
 	float mix[12] = {0};
-	mix[0] = volume / 100.0f;
-	mix[1] = mix[0];
-	ndspChnSetMix(0, mix);
-
-	ndspChnWaveBufAdd(ndsp_channel, &se_buf[se_channel]);
+	mix[0] = mix[1] = volume / 100.0f;
+	ndspChnSetMix(chan, mix);
```
Comment (carstene1ns) [inline src/platform/3ds/audio.cpp:282]: This is actually a second Bugfix (note channel 0 was used before).

File: src/platform/3ds/audio.cpp [hunk 2]
```diff
@@ -106,96 +141,91 @@ void CtrAudio::BGM_Pause()  {
...
+			chan = i;
+			//DebugLog("Using channel {} for SE {}", chan, se_cache->GetName());
 			break;
 		}
 	}
-
-	if (se_channel == -1) {
-		Output::Warning("Couldn't play SE {}: No free channel available", se->GetName());
+	if (chan == -1) {
+		Output::Warning("Couldn't play SE {}: No free channel available", se_cache->GetName());
```
Comment (fdelapena) [inline src/platform/3ds/audio.cpp:224]: Maybe there's good chance to output as debug to match other audio ports.

**Review by fdelapena — APPROVED** [review]
Battery status feature looks great, but I feel tones look a bit bright/saturated compared to the existing ones on the dark keypad design.

**Review by Ghabry — APPROVED** [review]
Thanks for the audio fixes.

I agree with fdelapenas nits but for homebrew we usually accept even "not 100% perfect" contributions. So would be fine if the nits are resolved in a follow up PR (though don't think anybody will complain soon xD)

**Comment by Ghabry** [comment]
Battery indicators provided by trailboss

battery2

**Comment by carstene1ns** [comment]
Ready to merge.

**Pull request merged**